File: DeepLearning/D2L/code/a.py
import math
def find_factors(n):
    factors = []
    for i in range(1, int(math.sqrt(n + 1)) ):

        if n % i == 0:
            factors.append(i)
            if n//i!=i : factors.append(n//i)
            
            print(i,n//i)
    return factors

# 示例使用
n = 1145141919
factors = find_factors(n)
print(f"数字 {n} 的所有因子是: {factors}")

File: DeepLearning/D2L/code/a.cpp
#include <iostream>
#include <vector>

std::vector<long long> findFactors(long long n) {
    std::vector<long long> factors;
    for (long long i = 1; i <= n; ++i) {
        if (n % i == 0) {
            factors.push_back(i);
        }
    }
    return factors;
}

int main() {
    long long n = 1145141919;
    std::vector<long long> factors = findFactors(n);

    std::cout << "The factors of " << n << " are: ";
    for (long long factor : factors) {
        std::cout << factor << " ";
    }
    std::cout << std::endl;

    return 0;
}
